### PR TITLE
Get Saml IDP metadata from file

### PR DESCRIPTION
**Pull request by edtubillara**
This PR makes the SAML integration more flexible. This PR allows setting a file source for the identity metadata instead of just using a URL. This also enables the auth signing request when the service provider redirects the user to the identity provider.

### CODE REVIEW

**Comment by nirajdp76** [comment]
ok to test

**Comment by lihkin213** [comment]
retest

File: roles/keystone/templates/etc/keystone/keystone.conf [hunk 1]
```diff
@@ -94,13 +94,6 @@ methods = external,password,token{{ ',saml2' if keystone.federation.enabled|bool
 {%- endfor -%}
 {%- endif %}
 
-{% if keystone.federation.enabled|bool and keystone.federation.sp.oidc.enabled|bool -%}
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/keystone.conf]: I am assuming this is not required anymore.

File: roles/keystone/templates/etc/keystone/keystone.conf [hunk 2]
```diff
@@ -115,7 +108,7 @@ remote_id_attribute = {{ sp.remote_id_attribute }}
 
 {% if keystone.federation.enabled|bool  and keystone.federation.sp.saml.enabled|bool -%}
 
-[saml]
+[saml2]
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/keystone.conf]: This changed in Newton?

File: roles/keystone/templates/etc/keystone/keystone.conf (hunk 1 above)
Comment (edtubillara) [inline roles/keystone/templates/etc/keystone/keystone.conf]: This should not be needed anymore.

File: roles/keystone/templates/etc/keystone/keystone.conf (hunk 2 above)
Comment (edtubillara) [inline roles/keystone/templates/etc/keystone/keystone.conf]: Yes, (or it was changed in Mitaka?), but we should be able to set saml2 plugin in the auth plugins and Keystone will automatically know it should be the Mapped class.

**Comment by lihkin213** [comment]
LGTM

File: roles/keystone/templates/etc/keystone/keystone.conf (hunk 2 above)
Comment (edtubillara) [inline roles/keystone/templates/etc/keystone/keystone.conf]: It is changed in Mitaka.

**Pull request merged**